examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```
@@ -0,0 +1,243 @@
+   * @param host Hostname that Spark Streaming context has to connect for receiving data.
+   * @param port Port that Spark Streaming context has to connect for receiving data.
+   * @param outputPath Output path to save the processed data read by the Spark Streaming context.
+   */
+  def processStreamingWordCount(
+      host: String,
+      port: Int,
+      outputPath: String): ScalaJobHandle[Unit] = {
+    scalaClient.submit { context =>
+      context.createStreamingContext(15000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyContent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(12000)
+      ssc.stop(false, true)
+    }
+  }
+
+  /**
+   * Submits a spark sql job to the livy server.
+   *
+   * The sql context job reads data frames from the given json path and executes
+   * a sql query to get the word with max count on the temp table created with data frames.
+   * @param inputPath Input path to the json data containing the words.
+   */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[String] = {
+    scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      try {
+        val rdd = sqlctx.read.json(inputPath)
+        rdd.registerTempTable("words")
+        val result = sqlctx.sql("select word, count(word) as word_count from words " +
+          "group by word order by word_count desc limit 1")
+        result.first().toString()
+      } catch {
+        case exception: IOException => {
+          throw new IOException("The input path does not have any data frame", exception)
```

nit: the exception could be for a completely different reason than what the message says. better to just let the exception propagate without trying to guess the reason.